plugins/soundsourcem4a/audiosourcem4a.cpp

```diff
@@ -0,0 +1,337 @@
+}
+
+AudioSource::diff_type AudioSourceM4A::seekSampleFrame(diff_type frameIndex) {
+    if (m_curFrameIndex != frameIndex) {
+        const MP4SampleId sampleBlockId =
+            kMinSampleBlockId + (frameIndex / kFramesPerSampleBlock);
+        if (!isValidSampleBlockId(sampleBlockId)) {
+            return m_curFrameIndex;
+        }
+        if ((frameIndex < m_curFrameIndex) || // seeking backwards?
+            !isValidSampleBlockId(m_curSampleBlockId) || // invalid seek position?
+            (sampleBlockId > (m_curSampleBlockId + kNumberOfPrefetchSampleBlocks))) { // jumping forward?
+            // Restart decoding one or more blocks of samples backwards
+            // from the calculated starting block to avoid audible glitches.
+            // Implementation note: The type MP4SampleId is unsigned so we
+            // need to be careful when subtracting!
+            if ((kMinSampleBlockId + kNumberOfPrefetchSampleBlocks) < sampleBlockId) {
+                m_curSampleBlockId = sampleBlockId - kNumberOfPrefetchSampleBlocks;
+            } else {
+                m_curSampleBlockId = kMinSampleBlockId;
+            }
+            NeAACDecPostSeekReset(m_hDecoder, m_curSampleBlockId);
+            m_curFrameIndex = (m_curSampleBlockId - kMinSampleBlockId) * kFramesPerSampleBlock;
+            // discard input buffer
+            m_inputBufferOffset = 0;
+            m_inputBufferLength = 0;
+        }
+        // decoding starts before the actual target position
+        DEBUG_ASSERT(m_curFrameIndex <= frameIndex);
+        const size_type prefetchFrameCount = frameIndex - m_curFrameIndex;
+        // prefetch (decode and discard) all samples up to the target position
+        DEBUG_ASSERT(frames2samples(prefetchFrameCount) <= m_prefetchSampleBuffer.size());
+        readSampleFrames(prefetchFrameCount, &m_prefetchSampleBuffer[0]);
+    }
+    DEBUG_ASSERT(m_curFrameIndex == frameIndex);
+    return frameIndex;
+}
+
+AudioSource::size_type AudioSourceM4A::readSampleFrames(
+        size_type numberOfFrames, sample_type* sampleBuffer) {
+    if (!isValidSampleBlockId(m_curSampleBlockId)) {
+        qWarning() << "Invalid MP4 sample block" << m_curSampleBlockId;
+        return 0;
+    }
+    sample_type* pSampleBuffer = sampleBuffer;
+    size_type numberOfFramesRemaining = numberOfFrames;
+    while (0 < numberOfFramesRemaining) {
+        DEBUG_ASSERT(m_inputBufferOffset <= m_inputBufferLength);
+        if (m_inputBufferOffset >= m_inputBufferLength) {
+            // reset input buffer
+            m_inputBufferOffset = 0;
+            m_inputBufferLength = 0;
+            if (isValidSampleBlockId(m_curSampleBlockId)) {
+                // fill input buffer with next block of samples
+                u_int8_t* pInputBuffer = &m_inputBuffer[0];
+                u_int32_t inputBufferLength = m_inputBuffer.size(); // in/out parameter
+                if (!MP4ReadSample(
+                    m_hFile, m_trackId, m_curSampleBlockId,
```

Comment:
Sorry for the annoyance. You code style was very OK before this commit.
I know that some people are upset by to much code styling comments. I have keeped pointing at these because, I got the impression that this was no problem for you and you like to have your code almost perfect as well.

Since code style is not that important over well working and commented code, we may find a new agreement which comment level you like. We have really more important things to do then reformat this code. 

But let me explain the issue: 
We reference the Google C++ style in our coding guidelines. 
Google uses 2 space indent for logical blocks and double indent for line breaks or alignment to the opening ( 
Since Mixxx has 4 space indent and also a preferred line length of 80, a 8 indent suits, compared to Google) more often better than the "(" alignment for line breaks.  

So we will have on special exceptional case: 
```
if ( looooooooooooooog ||
        looooooooong) {
   statement()
}
```
"(" alignement is not allowed her because it has a 4 indent which is reserved for code blocks.  

@rryan has tried to setup a code style checker which prefers "(" alignment, but he fails because of this exception.  

I use Eclipse Code Style "K&R" with only a single exception "Tap policy" = Space only. 
This will not produce "(" alignment, but 8 indent for line breaks, which is OK for me and Mixxx. 
I correct that manually in my code when it looks ugly, but that is my personal preference, not important here. 

So if you don't mind, revert this indent commit and the changes in your Auto formatter and I will stop complaining about indent (or continue, its up to you)  

I will also update our style guide with this information.